src/sources/util/unix_datagram.rs

```diff
@@ -36,6 +38,18 @@ pub fn build_unix_datagram_source(
         let socket = UnixDatagram::bind(&listen_path).expect("Failed to bind to datagram socket");
         info!(message = "Listening.", path = ?listen_path, r#type = "unix_datagram");
 
+        if let Some(socket_permission) = socket_file_mode {
+            match fs::set_permissions(&listen_path, fs::Permissions::from_mode(socket_permission)) {
+                Ok(_) => info!("Socket permissions updated to {:#o}.", socket_permission),
```

Comment:
```suggestion
                Ok(_) => info!("Socket permissions updated.", socket_permission),
```

Similar to below, we generally like to attach dynamic values separately and keep the messages static.

Reply:
I think we should err on the side of caution here though, and return an error if we cannot set permissions as this does open up a possible security hole. That would mean making the signature here a `Result<Source, ...>` and bubbling up the error so that Vector aborts.

It should be pretty rare to hit this in practice.

Reply:
> This would help avoid race conditions if the socket is renamed in the middle somehow.

Is it a supported usecase? I don't really see how it could happen

> and return an error if we cannot set permissions

I agree, I'll do that!

Reply:
Ah, sorry, I mean if some external process managed to rename the file in-between Vector creating it and setting the permissions. It's a pretty extreme edge-case, to be sure, but using `fchmod` should protect against it since it sets the permissions based on the file descriptor.

Reply:
okay so both issues should be good now :)